Sources/WhoopDIKit/Module/DependencyModule.swift

```diff
@@ -65,6 +65,48 @@ open class DependencyModule {
         }
     }
     
+    /// Defines an accumulation that is recalculated on each request.
+    /// Values are accumulated across the container hierarchy using the AccumulationKey's accumulate function.
+    /// - Parameters:
+    ///   - name: An optional name which can be used to disambiguate between multiple accumulations of the same type.
+    ///   - key: The AccumulationKey type that defines how values are accumulated
+    ///   - provideValue: A closure that provides the value to accumulate
+    public final func accumulateFactory<Key: AccumulationKey>(name: String? = nil, for key: Key.Type, provideValue: @escaping () throws -> Key.AccumulatedValue) {
+        dependencies.append(FactoryAccumulationDefinition(name: name, accumulationKey: key, valueProvider: { _ in try provideValue() }))
+    }
+
+    /// Defines an accumulation that is recalculated on each request with parameters.
+    /// Values are accumulated across the container hierarchy using the AccumulationKey's accumulate function.
+    /// - Parameters:
+    ///   - name: An optional name which can be used to disambiguate between multiple accumulations of the same type.
+    ///   - key: The AccumulationKey type that defines how values are accumulated
+    ///   - provideValue: A closure that provides the value to accumulate, with parameters
+    public final func accumulateFactoryWithParams<Key: AccumulationKey, Param>(name: String? = nil, for key: Key.Type, provideValue: @escaping (Param) throws -> Key.AccumulatedValue) {
```

Comment:
Let's avoid adding parameter variants. We have a better pattern for that use case now (local inject)